spectractor/extractor/spectrum.py

```diff
@@ -880,6 +880,7 @@ def calibrate_spectrum_with_lines(spectrum):
     if x0 is None:
         x0 = spectrum.target_pixcoords
         spectrum.x0 = x0
+    # MFL notes: the logic with D seems confused here
```

Comment:
D here is the initial guess for the distance between the disperser and the CCD. The function can work either if first we loaded an image or a spectrum fits file. In the first case the default paramerters.py value is read, otherwise the previously fitted D value written in the header is read.